### DISCUSSION TITLE
`limit_max_requests` is not working?

### DISCUSSION THREAD

**Discussion by lukasugar** [Kludex/uvicorn#2412] [Potential Issue]
I want to restart workers (or server) when a certain number of requests in hit.
I'm trying to use `limit_max_requests` for that.
Here's the code snippet:
```
if __name__ == "__main__":
    import uvicorn

    limit_max_requests = 5

    uvicorn.run(
        app, host="0.0.0.0", port=8000, limit_max_requests=limit_max_requests
    )
```

I can still submit many more requests than 5:
<img width="422" alt="image" src="https://github.com/user-attachments/assets/0ae757b1-6d01-4033-895f-fa6e973645e4">

So my questions are:
- Should `limit_max_requests` stop or restart it?
- Is there a way to verify that the server/worker has restarted?
- The reason why I need this is because some third party app has a memory leak. I want to restart because then the memory should be reset as well. Any ideas of a better approach? Fixing the third party library is not possible atm :) 

I'm using MacOS, and these packages:
```
python = ">3.11, <3.12"
uvicorn = "^0.23.2"
langserve = "^0.2.2"
python-decouple = "^3.8"
mypy = "^1.10.0"
poetry-dotenv-plugin = "^0.2.0"
python-dotenv = "^1.0.1"
langchain-openai = "^0.1.14"
langchain-core = "^0.2.11"
langgraph = "^0.1.5"
langchain = "^0.2.6"
pydantic = "<2"
aiosqlite = "^0.20.0"
```

**Comment by vvanglro** [comment]
When the request reaches the value of `limit_max_requests`, the child process will exit.

This value can be exceeded because the event loop cannot schedule the `on_tick` method to run in time when a large number of concurrent requests are received.

You can use uvicorn >= 0.30.0 https://github.com/encode/uvicorn/pull/2183, the new process manager restarts the process when the maximum request limit is reached (the child process exits and the new process manager start a child process).
